cmd/clusterctl/pkg/client/repository/repository_local_unix.go

```diff
@@ -0,0 +1,240 @@
+// path to the components yaml on the local filesystem.
+// In order to support different versions, the directories containing provider
+// specific data must adhere to the following layout:
+// {basepath}/{provider-name}/{version}/{components.yaml}
+//
+// (1): {provider-name} must match the value returned by Provider.Name()
+// (2): {version} must exactly obey the syntax and semantics of
+// the "Semantic Versioning" specification (http://semver.org/).
+//
+// Concrete example:
+// /home/user/go/src/sigs.k8s.io/aws/v0.4.7/infrastructure-components.yaml
+// basepath: /home/user/go/src/sigs.k8s.io
+// provider-name: aws
+// version: v0.4.7
+// component.yaml: infrastructure-components.yaml
+
+type localRepository struct {
+	providerConfig        config.Provider
+	configVariablesClient config.VariablesClient
+	basepath              string
+	providerName          string
+	defaultVersion        string
+	rootPath              string
+	componentsPath        string
+}
+
+var _ Repository = &localRepository{}
+
+// DefaultVersion returns defaultVersion field of localRepository struct
+func (r *localRepository) DefaultVersion() string {
+	return r.defaultVersion
+}
+
+// RootPath returns rootPath field of localRepository struct
+func (r *localRepository) RootPath() string {
+	return r.rootPath
+}
+
+// ComponentsPath returns componentsPath field of localRepository struct
+func (r *localRepository) ComponentsPath() string {
+	return r.componentsPath
+}
+
+// GetFile returns a file for a given provider version
+func (r *localRepository) GetFile(version, fileName string) ([]byte, error) {
+	var err error
+
+	if version == "latest" {
+		version, err = r.getLatestRelease()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get the latest release")
+		}
+	} else if version == "" {
+		version = r.defaultVersion
+	}
+
+	absolutePath := filepath.Join(r.basepath, r.providerName, version, r.rootPath, fileName)
+
+	if f, err := os.Stat(absolutePath); err == nil {
+		if f.IsDir() {
+			return nil, errors.Errorf("invalid path: file %q is actually a directory %q", fileName, absolutePath)
+		}
+		content, err := ioutil.ReadFile(absolutePath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to read files from local release %s", version)
+		}
+		return content, nil
+	}
+	return nil, errors.Errorf("failed to read files from local release %s, path %s", version, absolutePath)
+
+}
+
+// GetVersion returns the list of versions that are available for a local repository
+func (r *localRepository) GetVersions() ([]string, error) {
+	versions, err := r.getVersions()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get local repository versions")
+	}
+	return versions, nil
+}
+
+// newLocalRepository returns a localRepository implementation
+func newLocalRepository(providerConfig config.Provider, configVariablesClient config.VariablesClient) (*localRepository, error) {
+
+	var err error
+
+	urlSplit, err := url.Parse(providerConfig.URL())
```

Comment:
my bad, but I confused this line with 132. Would you be so kind to
urlSplit --> url
parts --> urlSplit

Reply:
no problem.